### PR TITLE
Added capability to use private images from gitlab

### PR DESCRIPTION
**Pull request by leafty**
Requires: https://github.com/SwissDataScienceCenter/renku/pull/301

### CODE REVIEW

File: src/notebooks_service.py [hunk 1]
```diff
@@ -170,18 +170,32 @@ def launch_notebook(user, namespace, project, commit_sha, notebook=None):
...
     # 1. launch using spawner that checks the access
+    payload = {
+        'branch': request.args.get('branch', 'master'),
+        'commit_sha': commit_sha,
+        'namespace': namespace,
+        'notebook': notebook,
+        'project': project,
+    }
+    if os.environ.get('GITLAB_REGISTRY_SECRET'):
+        payload['image_pull_secrets'] = payload.get('image_pull_secrets', [])
```
Comment (rokroskar) [inline src/notebooks_service.py]: `payload` here will never contain `image_pull_secrets` -- or is that something you expect in the future?

File: src/notebooks_service.py [hunk 2]
```diff
@@ -170,18 +170,32 @@ def launch_notebook(user, namespace, project, commit_sha, notebook=None):
...
+        payload['image_pull_secrets'].append(
+            os.environ['GITLAB_REGISTRY_SECRET']
+        )
+    app.logger.debug(
+        'POST {prefix}/users/{user[name]}/servers/{server_name}'.format(
+            prefix=auth.api_url, user=user, server_name=server_name
+        )
+    )
+    app.logger.debug(
+        'data = {}'.format(payload)
```
Comment (rokroskar) [inline src/notebooks_service.py]: probably don't want to log the image registry secret

File: src/notebooks_service.py (hunk 1 above)
Comment (leafty) [reply src/notebooks_service.py]: I thought it would make the code robust to changes anywhere.

Comment (rokroskar) [reply src/notebooks_service.py]: 👍 just making sure I'm not missing something obvious

File: src/notebooks_service.py (hunk 2 above)
Comment (leafty) [reply src/notebooks_service.py]: It's just the names of the secret like 'renku-notebooks-registry' so not really critical.

Comment (rokroskar) [reply src/notebooks_service.py]: ah true

**Pull request merged**